islands/VersionSelect.tsx

```diff
@@ -0,0 +1,33 @@
+/** @jsx h */
+import { h, tw } from "../deps.ts";
+
+export default function VersionSelect({ versions, selectedVersion }: {
+  versions: Record<string, string>;
+  selectedVersion: string;
+}) {
+  return (
+    <div class={tw`max-w-xs rounded-md shadow-sm w-full`}>
+      <select
+        class={tw
+          `block form-select w-full transition duration-150 ease-in-out sm:text-sm sm:leading-5`}
+        value={selectedVersion}
+        onChange={(e) => {
+          if (e.currentTarget.value !== selectedVersion) {
+            location.href = versions[e.currentTarget.value];
+          }
+        }}
+      >
+        {!Object.hasOwn(versions, selectedVersion) && (
+          <option key={selectedVersion} value={selectedVersion}>
+            {selectedVersion}
+          </option>
+        )}
+        {Object.keys(versions).map((tag) => (
```

Comment:
Objects don't have an ordering  - presumably we want the versions ordered. Maybe use `[string,string][]`instead?

Reply:
> Objects don't have an ordering

They do. Since ES2015 ordering of string keys in objects follow insertion order.